wordpress-mobile/WordPress-Android
# Add stock media source to media picker

Fixes #12641

This PR introduces the Stock media data source. Since the behaviour of the data source is different than the device list and the WP media library, it also introduces some changes to the MediaSource interface and the MediaLoader. The interface they use now is much simpler (with one simple `load` method). 

The behaviour of the stock media picker is that it starts empty (the empty screen is not yet implemented as it should be) with an always enabled `search` bar. Once you search a term longer than 2 letters, it searches for the items and populates the list. 

Since the stock media need to perform an upload operation once they are selected, I've introduced `MediaInsertUseCase` interface for a use cases that handles the upload. The `insert` method is called when the user clicks on `insert` and it shows a progress dialog. This could be used by any of the media sources. The default implementation doesn't do anything. 

I'll add more unit tests in the following PR

To test
- Make sure the consolidated picker flag is turned on
- Go to Media library
- Click on `+`
- Choose "Choose from Free Photo Library`
- Notice that the stock picker is opened
- Notice that the search view mode is on
- Notice that the search mode cannot be cancelled
- Search for something with 2 letters
- Nothing gets searched
- Search for something with 3 or more letters
- The results are loaded
- Select multiple items
- Confirm insert
- Notice the progress dialog is shown
- Notice it can be cancelled
- Let the progress dialog finish
- The screen gets closed and the images are added to your media library


To test 2
- Make sure the consolidated picker flag is turned on
- Go to GB editor/Add Image block/Add Media/Choose from Free Photo Library
- Search for something with >=3 letters
- Notice you can select only one item
- Insert it
- Notice the loading dialog
- Notice the item gets added to the GB image block


To test 3
- Check nothing is broken in the other data sources

PR submission checklist:
